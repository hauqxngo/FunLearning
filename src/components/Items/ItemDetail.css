.item-detail {
  cursor: pointer;
}

.img-hover-zoom {
  overflow: hidden;
}

.item-detail img {
  height: 200px;
  object-fit: cover;
  cursor: pointer;
  -webkit-transform: scale(1);
  transform: scale(1);
  -webkit-transition: 0.5s ease-in-out;
  transition: 0.5s ease-in-out;
}

.item-detail:hover img {
  -webkit-transform: scale(1.2);
  transform: scale(1.2);
}
