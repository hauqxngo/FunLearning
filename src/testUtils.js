import React from "react";
import UserContext from "./UserContext";

const testUser = {
  username: "testuser",
  first_name: "Test",
  last_name: "User",
  email: "test@test.com",
  photo_url: null,
};

const UserProvider = ({ children, currentUser = testUser }) => (
  <UserContext.Provider value={{ currentUser }}>
    {children}
  </UserContext.Provider>
);

export { UserProvider };
