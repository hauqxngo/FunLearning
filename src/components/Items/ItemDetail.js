import React from "react";
import { Button } from "reactstrap";
import { Card, CardBody, CardImg, CardTitle } from "reactstrap";
import "./ItemDetail.css";

/** Show information about a item.
 *
 * Is rendered by Items to show a "card" for each item.
 *
 * Receives view func prop from parent, which is called on view.
 *
 * ItemCard -> ItemDetail
 */

const ItemDetail = ({ id, name }) => {
  let audio = new Audio(`/sounds/${name}.mp3`);
  const start = () => {
    audio.play();
  };

  return (
    <div className="item-detail shadow">
      <Card>
        <div className="img-hover-zoom">
          <CardImg
            alt={name}
            src={`/images/items/${name}.jpeg`}
            onClick={start}
          ></CardImg>
        </div>
        <CardBody>
          <CardTitle className="text-success" tag="h4">
            {name}
          </CardTitle>
          <div className="d-grid gap-2 d-md-flex justify-content-md-end">
            <Button color="success me-md-2" onClick={start}>
              PLAY <i className="fas fa-volume-high"></i>
            </Button>
          </div>
        </CardBody>
      </Card>
    </div>
  );
};

export default ItemDetail;
