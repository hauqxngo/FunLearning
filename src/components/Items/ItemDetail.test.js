import React from "react";
import { render } from "@testing-library/react";
import ItemDetail from "./ItemDetail";
import { UserProvider } from "../../testUtils";

it("matches snapshot", () => {
  let item = { name: "Aa" };
  const { asFragment } = render(
    <UserProvider>
      <ItemDetail item={item} />
    </UserProvider>
  );
  expect(asFragment()).toMatchSnapshot();
});
