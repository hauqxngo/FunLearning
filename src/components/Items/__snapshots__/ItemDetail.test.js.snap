// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="item-detail shadow"
  >
    <div
      class="card"
    >
      <div
        class="img-hover-zoom"
      >
        <img
          class="card-img"
          src="/images/items/undefined.jpeg"
        />
      </div>
      <div
        class="card-body"
      >
        <h4
          class="text-success card-title"
        />
        <div
          class="d-grid gap-2 d-md-flex justify-content-md-end"
        >
          <button
            class="btn btn-success me-md-2"
            type="button"
          >
            PLAY 
            <i
              class="fas fa-volume-high"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
